### PR TITLE
When editing order and the item is unavailable, it doesn't shows as unavailable

### PR DESCRIPTION
**Pull request by vadym-hyryn**
https://dev.azure.com/orckestra001/OrckestraCommerce/_workitems/edit//

### CODE REVIEW

File: src/Orckestra.Composer.Cart/Services/Order/OrderHistoryViewService.cs [hunk 1]
```diff
@@ -569,6 +570,17 @@ public virtual async Task<CompleteCheckoutViewModel> SaveEditedOrderAsync(string
                 throw new InvalidOperationException($"Cannot save edited order #${orderNumber} as it doesn't exist.");
             }
 
+            var removeInvalidLineItemsParam = new RemoveInvalidLineItemsParam
+            {
+                CartName = order.Id.Replace("-", "").ToLower(),
+                CartType = CartConfiguration.OrderDraftCartType,
```
Comment (InnaBoitsun) [inline src/Orckestra.Composer.Cart/Services/Order/OrderHistoryViewService.cs:576]: try order.Id.ToString("N")

Comment (vadym-hyryn) [reply src/Orckestra.Composer.Cart/Services/Order/OrderHistoryViewService.cs:576]: Id is a string there, I think that will not work

File: src/Orckestra.Composer.Cart/Services/Order/OrderHistoryViewService.cs [hunk 2]
```diff
@@ -569,6 +570,17 @@ public virtual async Task<CompleteCheckoutViewModel> SaveEditedOrderAsync(string
...
+                CultureInfo = ComposerContext.CultureInfo,
+                CustomerId = ComposerContext.CustomerId,
+                Scope = ComposerContext.Scope,
+                BaseUrl = baseUrl
+            };
+            await CartService.RemoveInvalidLineItemsAsync(removeInvalidLineItemsParam);
```
Comment (InnaBoitsun) [inline src/Orckestra.Composer.Cart/Services/Order/OrderHistoryViewService.cs:582]: .ConfigureAwait(false)

File: src/Orckestra.Composer.Cart/Services/Order/OrderHistoryViewService.cs [hunk 3]
```diff
@@ -569,6 +571,17 @@ public virtual async Task<CompleteCheckoutViewModel> SaveEditedOrderAsync(string
                 throw new InvalidOperationException($"Cannot save edited order #${orderNumber} as it doesn't exist.");
             }
 
+            var removeInvalidLineItemsParam = new RemoveInvalidLineItemsParam
```
Comment (InnaBoitsun) [inline src/Orckestra.Composer.Cart/Services/Order/OrderHistoryViewService.cs:574]: this code is not actual here - you can remove it

File: src/Orckestra.Composer.Website/UI.Package/Typescript/Composer.Cart/OrderSummary/OrderSummaryController.ts [hunk 4]
```diff
@@ -89,6 +89,19 @@ module Orckestra.Composer {
...
+                        if (!nextStepUrl) {
+                            throw 'No next step Url was defined.';
+                        }
+
+                        this.Mode.Loading = true;
+                        self.orderSummaryService.cleanCart().done(() => {
+                            window.location.href = nextStepUrl;
+                        }, reason => {
+                            console.error('Error while removing invalid items from the cart', reason);
+                        });
```
Comment (InnaBoitsun) [inline src/Orckestra.Composer.Website/UI.Package/Typescript/Composer.Cart/OrderSummary/OrderSummaryController.ts:104]: we do not reload page here, please validate if cleanCart return cart model and you should have something like this
self.orderSummaryService.cleanCart().then(cart => .selfeventHub.publish(CartEvents.CartUpdated, { data: cart }))

**Review by InnaBoitsun — CHANGES_REQUESTED** [review]

File: src/Orckestra.Composer.Cart/Services/Order/OrderHistoryViewService.cs [hunk 5]
```diff
@@ -22,8 +22,8 @@
 using System.Linq;
 using System.Threading.Tasks;
 using Orckestra.Overture.ServiceModel.Orders.Fulfillment;
+using Orckestra.Composer.Cart.Extensions;
```
Comment (InnaBoitsun) [inline src/Orckestra.Composer.Cart/Services/Order/OrderHistoryViewService.cs:25]: do we need this changes?

File: src/Orckestra.Composer.Website/UI.Package/Typescript/Composer.Cart/OrderSummary/OrderSummaryController.ts [hunk 6]
```diff
@@ -89,6 +89,14 @@ module Orckestra.Composer {
                             console.error('Error while proceeding to Checkout', reason);
                             ErrorHandler.instance().outputErrorFromCode('ProceedToCheckoutFailed');
                         });
+                    },
+                    removeInvalidLineItems() {
+                        this.Mode.Loading = true;
+                        self.orderSummaryService.cleanCart()                
+                            .then(this.cartService.invalidateCache())
```
Comment (InnaBoitsun) [inline src/Orckestra.Composer.Website/UI.Package/Typescript/Composer.Cart/OrderSummary/OrderSummaryController.ts:96]: should be **self**.cartService.invalidateCache(); and move before the cleanCart()

File: src/Orckestra.Composer.Website/UI.Package/Typescript/Composer.Cart/OrderSummary/OrderSummaryController.ts [hunk 7]
```diff
@@ -89,6 +89,14 @@ module Orckestra.Composer {
...
+                            .then(cart => 
+                                this.eventHub.publish(CartEvents.CartUpdated, { data: cart })
```
Comment (InnaBoitsun) [inline src/Orckestra.Composer.Website/UI.Package/Typescript/Composer.Cart/OrderSummary/OrderSummaryController.ts:98]: should be **self**.eventHub

File: src/Orckestra.Composer.Website/UI.Package/Typescript/Composer.Cart/OrderSummary/OrderSummaryController.ts [hunk 8]
```diff
@@ -89,6 +89,14 @@ module Orckestra.Composer {
...
                             ErrorHandler.instance().outputErrorFromCode('ProceedToCheckoutFailed');
                         });
+                    },
+                    removeInvalidLineItems() {
+                        this.Mode.Loading = true;
+                        self.cartService.invalidateCache();
+                        self.orderSummaryService.cleanCart()
+                            .then(cart => 
+                                self.eventHub.publish(CartEvents.CartUpdated, { data: cart })
+                        );
```
Comment (InnaBoitsun) [inline src/Orckestra.Composer.Website/UI.Package/Typescript/Composer.Cart/OrderSummary/OrderSummaryController.ts:99]: .fin(() => this.Mode.Loading = false);

**Review by InnaBoitsun — APPROVED** [review]

**Pull request merged**